### PR TITLE
EQL: Give a name to all toml tests and enforce the naming of new tests

### PR DESCRIPTION
**Pull request by astefan**
This PR gives a name to all toml tests to a more easy tracking of failures while testing/debugging. Spin-off from https://github.com/elastic/elasticsearch/pull/59200.

### CODE REVIEW

File: x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml [hunk 1]
```diff
@@ -59,60 +68,72 @@ expected_event_ids  = [5]
...
 
 
 [[queries]]
+name = "numberStringConversion5"
 query = 'any where number(string(serial_event_id), 16) == 17'
 expected_event_ids = [11]
 
 
 [[queries]]
+name = "mathcWithCharacterClasses1"
```
Comment (matriv) [inline x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml:136]:
```suggestion
name = "matchWithCharacterClasses1"
```

File: x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml [hunk 2]
```diff
@@ -122,12 +143,14 @@ process where match(command_line, ?'.*?net1[ ]+localgroup.*?')
...
 
 [[queries]]
+name = "matchLiteAdditional"
 expected_event_ids  = [98]
 query = '''
 process where matchLite(command_line, ?'.*?net1.*?')
 '''
 
 [[queries]]
+name = "mathcWithCharacterClasses2"
```
Comment (matriv) [inline x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml:153]:
```suggestion
name = "matchWithCharacterClasses2"
```

File: x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml [hunk 3]
```diff
@@ -2,55 +2,64 @@
...
 
 [[queries]]
+name = "betweenAdditional2"
 expected_event_ids = [95]
 query = '''
 file where between(file_path, "dev", ".json", true) == "\\TestLogs\\something"
 '''
 
 [[queries]]
+name = "cidrMatchAdditional1"
```
Comment (matriv) [inline x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml:19]: why here is `additional` and next one is `stringCidrMatch` ?
Personally I would just name them `stringCidrMatchX` without the `Additional`.

File: x-pack/plugin/eql/qa/common/src/main/resources/test_queries_unsupported.toml [hunk 4]
```diff
@@ -8,12 +8,8 @@
 # in order to allow at least one round-trip test with the test harness.
 # This will be removed once the EQL implementation is wired and actually supports this query.
 
-
-[[queries]]
-expected_event_ids = []
-query = 'process where missing_field != null'
-
 [[queries]]
+name = "twoSequencesWithBys"
```
Comment (matriv) [inline x-pack/plugin/eql/qa/common/src/main/resources/test_queries_unsupported.toml:12]: I would rename `Bys` to `Keys`

File: x-pack/plugin/eql/qa/common/src/main/resources/test_queries_unsupported.toml [hunk 5]
```diff
@@ -241,26 +266,30 @@ expected_event_ids  = [78]
...
 [[queries]]
+name = "twoINsCaseInsensitive"
 case_insensitive = true
 query = '''
 process where opcode in (1,3) and process_name in (parent_process_name, "SYSTEM")
 '''
 expected_event_ids  = [2, 50, 51]
 
 [[queries]]
+name = "simeplTailWithSort"
```
Comment (matriv) [inline x-pack/plugin/eql/qa/common/src/main/resources/test_queries_unsupported.toml:292]:
```suggestion
name = "simpleTailWithSort"
```

**Review by matriv — APPROVED** [review]
LGTM. Nice enhancement to ease tracing failures.
Left some comments regarding naming and typos.

**Review by costin — APPROVED** [review]
LGTM - this must have been quite an effort.
I've encountered the same issue myself.

@rw-access can you please incorporate the changes back and see whether the names can be improved to better convey their intent? WithField1 doesn't say much vs mathWithDivision for example?

**Pull request merged**